pkg/sysfsnet/interfaces_linux.go

```diff
@@ -39,7 +41,21 @@ func Interfaces() ([]Interface, error) {
 	}
 
 	for _, dentry := range dents {
-		hwText, err := readMACFromFile(filepath.Join(sysClassNet, dentry.Name(), "address"))
+
+		entryPath := filepath.Join(sysClassNet, dentry.Name())
+
+		// os.Stat to follow symlinks and return the info of the target
+		info, err := os.Stat(entryPath)
+		if err != nil {
+			continue
```

Comment:
I think we should at least log `err` as a warning. That may be other reasons why `os.Stat(entryPath)` could fail, and we don't want it to always fail silently.

Reply:
@ganawaj would you mind improving that?